From 1c6ae26c7ab2a8943f8e4126cd21fe3bf20a25e4 Mon Sep 17 00:00:00 2001
From: Daniel Bast <2790401+dbast@users.noreply.github.com>
Date: Sun, 28 Dec 2025 11:35:04 +0100
Subject: [PATCH] Enable setting current_time from env variable

---
 tools/ddrbin_tool.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tools/ddrbin_tool.py b/tools/ddrbin_tool.py
index 745a5006..d185858d 100755
--- a/tools/ddrbin_tool.py
+++ b/tools/ddrbin_tool.py
@@ -2563,7 +2563,11 @@ def ddrbin_tool(argc, argv):
         else:
             if verinfo_editable == '':
                 #print(f"position_1={position_1}, position_2={position_2}, {old_verinfo_editable}")
-                current_time = datetime.now()
+                source_date_epoch = os.environ.get('SOURCE_DATE_EPOCH', '')
+                if source_date_epoch:
+                    current_time = datetime.fromtimestamp(int(source_date_epoch))
+                else:
+                    current_time = datetime.now()
                 verinfo_editable = current_time.strftime("%y/%m/%d-%H:%M.%S")
             if len(verinfo_editable) < verinfo_editable_length:
                 verinfo_editable = verinfo_editable.ljust(verinfo_editable_length)
-- 
2.55.0

